### PR TITLE
add hidden config with advanced settings

### PR DESCRIPTION
**Pull request by jerabekjak**

### CODE REVIEW

File: smoderp2d/providers/base/__init__.py [hunk 1]
```diff
@@ -112,6 +113,23 @@ def add_logging_handler(handler, formatter=None):
         handler.setFormatter(formatter)
         Logger.addHandler(handler)
 
+    def __load_hidden_config(self):
+        # load hidden configuration with advanced settings 
+        _path = os.path.join(os.path.dirname(__file__), '../../.config.txt')
```
Comment (landam) [inline smoderp2d/providers/base/__init__.py:118]:
```suggestion
        _path = os.path.join(os.path.dirname(__file__), '..', '..', '.config.txt')
```

Comment (landam) [reply smoderp2d/providers/base/__init__.py:118]: I would suggest to use `.config.ini` as filename

File: smoderp2d/.config.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[outputs]
+# yes or no
+extraout = no
```
Comment (landam) [inline smoderp2d/.config.txt:3]: Detailed description to be added.

File: smoderp2d/providers/base/__init__.py [hunk 3]
```diff
@@ -112,6 +113,23 @@ def add_logging_handler(handler, formatter=None):
...
+        if not os.path.exists(_path):
+            raise ConfigError("{} does not exist".format(
+                _path
+            ))
+
+        config = ConfigParser()
+        config.read(_path)
+
+        if not(config.has_option('outputs', 'extraout')):
```
Comment (landam) [inline smoderp2d/providers/base/__init__.py:127]:
```suggestion
        if not config.has_option('outputs', 'extraout'):
```

File: smoderp2d/providers/base/__init__.py (hunk 1 above)
Comment (jerabekjak) [reply smoderp2d/providers/base/__init__.py]: OK. changed it in commit 2884132d459968022ad277607eb47688595c56c3

File: smoderp2d/.config.txt (hunk 2 above)
Comment (jerabekjak) [reply smoderp2d/.config.txt]: see 70e2c7aaa96b36bfdb442cf3bb347d505ee447ff

**Comment by pesekon2** [comment]
@jerabekjak  Can this branch be deleted now?

**Pull request merged**